common/predictive-text/worker/correction/context-tracker.ts

```diff
@@ -351,7 +351,10 @@ namespace correction {
       }
 
       const hasDistribution = transformDistribution && Array.isArray(transformDistribution);
-      const primaryInput = hasDistribution ? transformDistribution[0].sample : null;
+      let primaryInput = hasDistribution ? transformDistribution[0].sample : null;
+      if(primaryInput && primaryInput.insert == "" && primaryInput.deleteLeft == 0 && primaryInput.deleteRight == 0) {
+        primaryInput = null;
+      }
       const isBackspace = primaryInput && primaryInput.insert == "" && primaryInput.deleteLeft > 0;
```

Comment:
```suggestion
      const isBackspace = primaryInput && primaryInput.insert == "" && primaryInput.deleteLeft > 0 && primaryInput.deleteRight == 0;
```
Not sure if this is necessary but seems like it might be important in some scenarios?